presentation/components/src/test/common/UnifiedSelectionContext.test.tsx

```diff
@@ -117,11 +120,61 @@ describe("UnifiedSelectionContext", () => {
   describe("context", () => {
     const keys = new KeySet();
 
-    it("gets current selection", () => {
-      const { result } = renderUnifiedSelectionContextHook(testIModel);
-      const stub = sinon.stub(Presentation.selection, "getSelection").returns(keys);
-      result.current.getSelection(10);
-      expect(stub).to.have.been.calledOnceWithExactly(testIModel, 10);
+    describe("getSelection", () => {
+      let stubGetSelection: sinon.SinonStub<[IModelConnection, number?], Readonly<KeySet>>;
+
+      beforeEach(() => {
+        stubGetSelection = sinon.stub(Presentation.selection, "getSelection").returns(keys);
+      });
+
+      it("gets current selection", () => {
+        const { result } = renderUnifiedSelectionContextHook(testIModel);
+        result.current.getSelection(10);
+        expect(stubGetSelection).to.have.been.calledOnceWithExactly(testIModel, 10);
+      });
+
+      it("makes KeySet reference be different from global KeySet", () => {
+        const { result } = renderUnifiedSelectionContextHook(testIModel);
+        const returnedKeySet = result.current.getSelection();
+        expect(returnedKeySet).not.to.be.equal(keys);
+      });
+
+      it("returns same KeySet reference for same selection level", () => {
+        const { result } = renderUnifiedSelectionContextHook(testIModel);
+        const firstKeySet = result.current.getSelection(10);
+        const secondKeySet = result.current.getSelection(10);
+        expect(firstKeySet).to.be.equal(secondKeySet);
+      });
+
+      it("returns different KeySet reference for different selection level", () => {
+        const { result } = renderUnifiedSelectionContextHook(testIModel);
+        const firstKeySet = result.current.getSelection(10);
+        const secondKeySet = result.current.getSelection(9);
+        expect(firstKeySet).not.to.be.equal(secondKeySet);
+      });
+
+      it("returns different KeySet reference after selection changes", () => {
+        const { result } = renderUnifiedSelectionContextHook(testIModel);
+        const firstKeySet = result.current.getSelection();
+
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises

```

Comment:
I see this in multiple places in this file, but not sure why it's needed. The `addToSelection` function is sync, so definitely not for that. Is `act` async? Should we `await` instead of ignoring the warning?

Reply:
`act` is async only when the callback is async. When I `await` this call, I get a runtime warning: `Warning: Do not await the result of calling act(...) with sync logic, it is not a Promise.`

Reply:
That's weird. I see some other tests using `act` without `await` or `eslint-disable-next-line`. 

Also, I see `react-test-renderer` types contain these:
```ts
export function act(callback: () => Promise<void | undefined>): Promise<undefined>;
export function act(callback: () => void | undefined): DebugPromiseLike;

// Intentionally doesn't extend PromiseLike<never>.
// Ideally this should be as hard to accidentally use as possible.
export interface DebugPromiseLike {
    // the actual then() in here is 1-ary, but that doesn't count as a PromiseLike.
    then(onfulfilled: (value: never) => never, onrejected: (reason: never) => never): never;
}
```
and it looks like eslint shouldn't complain about missing `awaits` for the sync version.

Reply:
Other tests that use `act` without `eslint` errors seem to be importing `act` from `@testing-library/react`.